src/InputOutput/IOstrings/IOstrings_outpath_name.jl

```diff
@@ -16,18 +18,22 @@ function IOstrings_outpath_name(problem_name, grid_resolution)
     # grid_resolution[2]   = Δy
     # grid_resolution[end] = Δz
     #
-    ndim = length(grid_resolution)
+    mpirank = MPI.Comm_rank(MPI.COMM_WORLD)
+    if mpirank == 0
+        ndim = length(grid_resolution)
 
-    outpath_string = string(grid_resolution[1], "mx")
-    for i = 2:ndim
-        ds = grid_resolution[i]
-        outpath_string = string(outpath_string, ds, "mx")
-    end
-    current_time = string(Dates.format(convert(Dates.DateTime, Dates.now()), Dates.dateformat"yyyymmdd_HHMMSS"))
-    OUTPATH = string("./output/",problem_name, "/", outpath_string,"_", current_time)
+        outpath_string = string(grid_resolution[1], "mx")
+        for i = 2:ndim
+            ds = grid_resolution[i]
+            outpath_string = string(outpath_string, ds, "mx")
+        end
+        current_time = string(Dates.format(convert(Dates.DateTime, Dates.now()), Dates.dateformat"yyyymmdd_HHMMSS"))
+        OUTPATH = joinpath("./output/",problem_name,outpath_string, "_", current_time) #string("./output/",problem_name, "/", outpath_string,"_", current_time)
```

Comment:
```suggestion
        outpath = joinpath("output", problem_name, string(outpath_string, "_", current_time))
```

Reply:
the "./" is needed so that it doesn't attempt to write to the root.

Reply:
`./output` and `output` are equivalent. root would be `/output`

Reply:
Alright I'll update it as per your suggestion and do a push